pilot/pkg/bootstrap/server.go

```diff
@@ -586,9 +586,10 @@ func (s *Server) initSecureDiscoveryService(args *PilotArgs, port string) error
 		return err
 	}
 
+	// TODO(myidpt): Be able to load custom CA certificates to verify clients.
 	cfg := &tls.Config{
 		GetCertificate: s.getIstiodCertificate,
-		ClientAuth:     tls.VerifyClientCertIfGiven,
+		ClientAuth:     tls.NoClientCert,
```

Comment:
I suppose this is fine since we use our own authentication further down the call that Costin added?

Reply:
although your description seems wrong that it "has no effect"?

Reply:
Yes you are right. We defer the authentication to later. Updated.

Reply:
Actually VerifyClientCertIfGiven must be present in order for the later authenticator to use the authenticated identity.

Reply:
I'm not sure this is correct - it will break the authentication using mTLS.